src/Repository/DatenweitergabeRepository.php

```diff
@@ -14,77 +17,65 @@
-            ->andWhere('a.activ = 1')
-            ->setParameter('teamPath', $teamPath)
-            ->getQuery()
-            ->getResult()
-            ;
+        $queryBuilder = $this->getBaseQueryBuilder($team);
+        return $queryBuilder->getQuery()->getResult();
     }
 
     /**
-     * @param array $teamPath
+     * @param Team $team
      * @return int|mixed|string
      * find transfers of type Datenweitergabe
      */
-    public function findActiveTransfersByTeamPath(array $teamPath): mixed
+    public function findActiveTransfersByTeam(Team $team): mixed
     {
-        return $this->createQueryBuilder('a')
-            ->andWhere('a.team IN (:teamPath)')
-            ->andWhere('a.activ = 1')
-            ->andWhere('a.art = 1')
-            ->setParameter('teamPath', $teamPath)
-            ->getQuery()
-            ->getResult()
-            ;
+        $queryBuilder = $this->getBaseQueryBuilder($team);
+        $queryBuilder->andWhere('a.art = 1');
+        return $queryBuilder->getQuery()->getResult();
     }
 
     /**
-     * @param array $teamPath
+     * @param Team $team
      * @return int|mixed|string
      * find transfers of type Auftragsverarbeitung
      */
-    public function findActiveOrderProcessingsByTeamPath(array $teamPath): mixed
+    public function findActiveOrderProcessingsByTeam(Team $team): mixed
     {
-        return $this->createQueryBuilder('a')
-            ->andWhere('a.team IN (:teamPath)')
-            ->andWhere('a.activ = 1')
-            ->andWhere('a.art = 2')
-            ->setParameter('teamPath', $teamPath)
-            ->getQuery()
-            ->getResult()
-            ;
+        $queryBuilder = $this->getBaseQueryBuilder($team);
+        $queryBuilder->andWhere('a.art = 2');
+        return $queryBuilder->getQuery()->getResult();
     }
 
-    public function findActiveByTeamAndUser($team, $user)
+    public function findActiveByTeamAndUser(Team $team, User $user)
     {
-        return $this->createQueryBuilder('a')
-            ->andWhere('a.team = :team')
+        $queryBuilder = $this->getBaseQueryBuilder($team);
+        $queryBuilder
             ->andWhere('a.assignedUser = :user')
+            ->setParameter('user', $user)
+        ;
+        return $queryBuilder->getQuery()->getResult();
+    }
+
+    private function getBaseQueryBuilder(Team $team) :QueryBuilder
+    {
+        $teamPath = $this->teamRepository->getPath($team);
+
+        return $this->createQueryBuilder('a')
+            ->innerJoin('a.verfahren', 'process')
+            ->andWhere('a.team = :team OR process.inherited = 1 AND process.activ = 1 AND process.team IN (:teamPath)')
             ->andWhere('a.activ = 1')
+            ->setParameter('teamPath', $teamPath)
             ->setParameter('team', $team)
-            ->setParameter('user', $user)
             ->orderBy('a.createdAt', 'DESC')
-            ->getQuery()
-            ->getResult();
+            ;
     }
```

Comment:
This query seems not to work properly.
When I create a new Datenweitergabe at `/datenweitergabe/new` and I go back to list at `/datenweitergabe` then the new entry is not listed despite it was created successfully.

Seems like only Datenweitergabe, which is currently not set in any vvt is affected by this problem i.e. where `a.verfahren` is still null. When the Datenweitergabe is connected to a vvt, then it shows up in the list.

Same problem occurs for policies, toms, etc.

Reply:
That should have been leftjoin, not innerjoin - my bad :see_no_evil: